Community Toolbox/scripts/utils_CommunityToolboxStruct/utils_CommunityToolboxStruct.gml

```diff
@@ -0,0 +1,16 @@
+/// @func struct_assign(destination,[...sources])
+/// @desc Assigns properties from one or more source structs to a destination struct, and returns the destination struct.
+/// @arg {Struct} destination       The struct to assign properties to.
+/// @arg {Struct} [...sources]      The source structs to get properties values from.
+/// @returns {Struct}
+function struct_assign(_destination) {
+    var _assign_method = method(_destination, function(_key, _value) {
```

Comment:
At one point, Game Maker recompiled `function()` declarations every time it saw them in a function body. I'm not sure if that's still the case, but if it is, then the following would be more efficient

```
static _assign_function = function(_key, _value) { self[$ _key] = _value }
var _assign_method = method(_destination, _assign_function)
```

since the function only gets compiled once now.